src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -0,0 +1,584 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage_clients/abs_client.h"
+
+#include "cloud_storage_clients/abs_error.h"
+#include "cloud_storage_clients/configuration.h"
+#include "cloud_storage_clients/logger.h"
+#include "cloud_storage_clients/util.h"
+#include "vlog.h"
+
+#include <utility>
+
+namespace {
+constexpr boost::beast::string_view text_plain = "text/plain";
+constexpr boost::beast::string_view block_blob = "BlockBlob";
+
+cloud_storage_clients::abs_client::list_bucket_result
+iobuf_to_list_bucket_result(iobuf buf) {
+    using namespace cloud_storage_clients;
+
+    try {
+        abs_client::list_bucket_result result;
+        auto root = util::iobuf_to_ptree(std::move(buf), abs_log);
+        auto enum_results = root.get_child("EnumerationResults");
+
+        if (auto prefix = enum_results.get_child_optional("Prefix"); prefix) {
+            result.prefix = prefix->get_value<ss::sstring>();
+        }
+
+        if (auto marker = enum_results.get_child_optional("Marker"); marker) {
+            result.is_truncated = true;
+        } else {
+            result.is_truncated = false;
+        }
+
+        auto blobs = enum_results.get_child("Blobs");
+        for (auto& [tag, blob] : blobs) {
+            abs_client::list_bucket_item item;
+            item.key = blob.get<ss::sstring>("Name");
+            item.size_bytes = blob.get<size_t>("Properties.Content-Length");
+            item.etag = blob.get<ss::sstring>("Properties.Etag");
+            // TODO(vlad): Why doesn't ss::sstring work here?
+            const auto last_modified = blob.get<std::string>(
+              "Properties.Last-Modified");
+            item.last_modified = util::parse_timestamp(last_modified);
+
+            result.contents.push_back(std::move(item));
+        }
+
+        return result;
+    } catch (...) {
+        vlog(
+          abs_log.error, "!!error parse result {}", std::current_exception());
+        throw;
+    }
+}
+} // namespace
+
+namespace cloud_storage_clients {
+
+static abs_rest_error_response
+parse_rest_error_response(boost::beast::http::status result, iobuf&& buf) {
```

Comment:
nit: iobuf is already move-only and fairly light-weight. there is often no reason to take it by r-value ref.

Reply:
Done in [force-push](https://github.com/redpanda-data/redpanda/compare/b8dace186bd904cfb634d13675389bc514eb6759..56c2f4b09eb678e762a99057b07de7756ae8ee8d).